llvm/lib/Target/ARM/Disassembler/ARMDisassembler.cpp

```diff
@@ -6491,7 +6478,6 @@ DecodeStatus ARMDisassembler::getThumbInstruction(MCInst &MI, uint64_t &Size,
                                STI);
     if (Result != MCDisassembler::Fail) {
       Size = 4;
-      Check(Result, AddThumbPredicate(MI));
```

Comment:
I'm not sure about this change.
`AddThumbPredicate` not only adds a predicate operand, but also advances IT/VPT block states.

Reply:
I think I've figure it out. I should call `UpdateThumbVFPPredicate` here instead.
Not sure about other tables I mentioned, but this should preserve the existing behavior for `NEONDup`.